torch/csrc/jit/passes/fold_conv_bn.cpp

```diff
@@ -251,29 +251,28 @@ void FoldConvBatchNormHelper::analyze(
 
       GRAPH_DEBUG("number of Conv-BatchNorm matches: ", matches.size());
       Graph* g = method.graph().get();
-      if (!conv_bn_names_.count(g)) {
+      if (!conv_bn_paths_.count(g)) {
         // This is to make sure we don't visit one graph multiple times
-        conv_bn_names_[g] = {};
+        conv_bn_paths_[g] = {};
         for (const Match& match : matches) {
+          if (!std::all_of(
```

Comment:
Instead of filtering the matches here, can we change the pattern definitions?
```c++
 const PatternInfo pattern2d = PatternInfo::parse_from_str(
      R"(	   
graph(%self, %input, %conv, %batchnorm):	
    %conv_out = prim::CallMethod[name="forward"](%conv, %input)	   
    %bn_out = prim::CallMethod[name="forward"](%batchnorm, %conv_out)	
    return (%bn_out))",	 
      {is_conv2d_module, is_batchnorm2d_module});	   

```
So that the module type is also checked? Currently the pattern seems to check for two forwards with the output of the first forward going as input to the second, which is generic.

Reply:
ClassType definition is not supported in subgraph matcher right now, the filtering will make sure we only match desired types